src/julia_internal.h

```diff
@@ -119,6 +119,11 @@ static inline void jl_assume_(int cond)
 #  define JL_USE_IFUNC 0
 #endif
 
+#ifndef JL_UV_LOCK
+#define JL_UV_LOCK() do {} while(0)
+#define JL_UV_UNLOCK() do {} while(0)
```

Comment:
Could add locks here guarded by `#ifdef JULIA_ENABLE_THREADING`.

Reply:
Can this simply use `jl_mutex_t`? It'll be much more clear which lock is involved, where it is defined and not having to reimplement everything to avoid GC infinite wait.

Reply:
I'd like to add real locks when this PR will be merged, to test it separately